exporter/awscloudwatchlogsexporter/exporter.go

```diff
@@ -214,6 +222,16 @@ func logToCWLog(resourceAttrs map[string]any, log plog.LogRecord, config *Config
 		body.Attributes = attrsValue(log.Attributes())
 		body.Resource = resourceAttrs
 
+		// scope should have a name at least
+		if scope.Name() != "" {
+			scopeBody := &scopeCwLogBody{
+				Name:       scope.Name(),
+				Version:    scope.Version(),
+				Attributes: attrsValue(scope.Attributes()),
+			}
+			body.Scope = scopeBody
+		}
+
 		bodyJSON, err = json.Marshal(body)
```

Comment:
Reading this section of code made me wonder what would happen there was an attribute with `Double` type and `Math.Inf()` value. 

With slightly modification to your PR I saw that the JSON marshaling would fail.

```
func testScope() pcommon.InstrumentationScope {
	scope := pcommon.NewInstrumentationScope()
	scope.SetName("test-scope")
	scope.SetVersion("1.0.0")
	scope.Attributes().PutStr("scope-attr", "value")
	scope.Attributes().PutDouble("test-inf", math.Inf(0))
	return scope
}


=== RUN   TestLogToCWLog
=== RUN   TestLogToCWLog/basic
    exporter_test.go:249: logToCWLog() error = json: unsupported value: +Inf, wantErr false
--- FAIL: TestLogToCWLog (0.00s)
    --- FAIL: TestLogToCWLog/basic (0.00s)
```

We have seen this JSON marshaling failure before in the EMF Exporter when a metric is processed with `Inf` or `NaN` values. In the EMF Exporter we completely drop the metric. 

I think this "issue" is outside the scope of the PR but should at least be noted somewhere. At minumum I think this affects the `awsemfexporter` and `awscloudwatchlogs` exporter. I think this would also affect any exporter that does JSON marshaling before export. Having to clean a set of attributes for each log seems computationally expensive so we should verify the cost of doing so before moving forward with a very simple fix (checking for and removing all NaN/Inf) values.